melodies_monet/plots/ozone_sonder_plots.py

```diff
@@ -0,0 +1,230 @@
+
+            colors=[fill_color_list[i+1]]*(len(altitude_threshold_list)-1)
+            for patch,color in zip(bplot_model['boxes'],colors):
+                patch.set_edgecolor(color)
+
+        plt.title('Comparison at '+str(station_name[0])+' on '+str(release_time)+' UTC')
+        plt.legend()
+        plt.ylim(altitude_range[0],altitude_range[1])
+        plt.xlim(ozone_range[0],ozone_range[1])
+        plt.ylabel('ALT-sea level (km)')
+        plt.xlabel('O3 (ppbv)')
+    elif altitude_method[0] == 'ground level':
+        output_list_obs = split_by_threshold(O3_OBS,ALT_gl,altitude_threshold_list)
+        bplot_obs=ax.boxplot(output_list_obs,vert = False,**boxplot_kwargs,positions=position_list_mid)
+        #add a line plot
+        ax.plot(O3_OBS,ALT_gl,'-k*',label = 'OBS:'+model_name_list[0])
+        for i in range(len(O3_MODEL_ALL)):
+            output_list_model = split_by_threshold(O3_MODEL_ALL[i],ALT_gl,altitude_threshold_list)
+            bplot_model=ax.boxplot(output_list_model,vert = False,**boxplot_kwargs,positions=position_list_mid)
+            #add a line plot
+            ax.plot(O3_MODEL_ALL[i],ALT_gl,'-*',color=fill_color_list[i+1],label = 'MODEL:'+model_name_list[i+1])
+            
+            colors=[fill_color_list[i+1]]*(len(altitude_threshold_list)-1)
+            for patch,color in zip(bplot_model['boxes'],colors):
+                patch.set_edgecolor(color)
+
+        plt.title('Comparison at '+str(station_name[0])+' on '+str(release_time)+' UTC')
+        plt.legend()
+        plt.ylim(altitude_range[0],altitude_range[1])
+        plt.xlim(ozone_range[0],ozone_range[1])
+        plt.ylabel('ALT-ground level (km)')
+        plt.xlabel('O3 (ppbv)')
+
+
+def split_by_threshold(o3_list_input,alt_list_input,threshold_list_input):
+    df = pd.DataFrame(data={'o3':o3_list_input,'alt':alt_list_input})
+    output_list = []
+    for i in range(1,len(threshold_list_input)):
+        df_here = df.o3.loc[(df.alt>threshold_list_input[i-1])&(df.alt<=threshold_list_input[i])]
+        output_list.append(df_here.values)
+    return output_list
+
+
+def density_scatter_plot_os(df,altitude_range,ozone_range,station_name,altitude_method,cmap_method):
+    #release height info,get height of each release site to be substract    
+    df_height = pd.DataFrame({
+         'station':['Boulder, Colorado','Huntsville, Alabama','University of Rhode Island','Trinidad Head, California'],
+         'height':[1.743,0.203,0.021,0.046]
+         })
+    if altitude_method[0] == 'ground level':
+        height_value = df_height.loc[df_height['station']==station_name[0]].values[0][1]
+    elif altitude_method[0] == 'sea level':
+        height_value = 0
+
+    #get o3 model, o3 sonder (obs) and height
+    df_short = df[df['altitude']<altitude_range[1]+height_value]
+    ALT = df_short['altitude']-height_value
+    O3_OBS = df_short['o3']
```

Comment:
You could add obsvar and modvar in line above and below to keep this more variable agnostic. Then you can use the same code for the met analysis too.

Reply:
this is done